src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  FC, MouseEvent, useState, useCallback, useMemo, ChangeEvent, memo,
+} from 'react';
+import classNames from 'classnames';
+import { Person } from '../../types/Person';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person) => void;
+  selectedPerson: Person | null;
+};
+
+const debounce = (callback: (...args: any) => void, delay: number) => {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+};
+
+export const Dropdown: FC<Props> = memo((props) => {
+  const { people, onSelected, selectedPerson } = props;
+
+  const [query, setQuery] = useState<string>(selectedPerson?.name || '');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+
+  const handleSelectPerson = (
+    event: MouseEvent<HTMLAnchorElement>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+
+    onSelected(person);
+    setQuery(person.name);
+    setAppliedQuery('');
+  };
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleChangeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    const currentQuery = event.target.value;
+
+    setQuery(currentQuery);
+    applyQuery(currentQuery.toLowerCase());
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person => {
+      return person.name.toLowerCase().includes(appliedQuery);
+    });
+  }, [appliedQuery]);
+
+  return (
+    <div className={classNames('dropdown', {
+      'is-active': appliedQuery,
+    })}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={handleChangeQuery}
+        />
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        <div className="dropdown-content">
+          {filteredPeople.length > 0 ? (
```

Comment:
You also have to hide options on text change

*hide suggestions on text change;*